### PR TITLE
UAR-1957: Fixing more issues at how data is reloaded when tabs are switched

### PR DESCRIPTION
**Pull request by nataliac06**

### CODE REVIEW

File: src/main/java/edu/arizona/kra/irb/onlinereview/CustomProtocolOnlineReviewServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ */
+public class CustomProtocolOnlineReviewServiceImpl extends ProtocolOnlineReviewServiceImpl implements ProtocolOnlineReviewService {
+    private static final Logger LOG = LoggerFactory.getLogger(CustomProtocolOnlineReviewServiceImpl.class);
+
+    public boolean isProtocolInStateToBeReviewed(ProtocolBase protocol) {
+        boolean isReviewable = false;
+        ProtocolSubmission submission = ((Protocol)protocol).getProtocolSubmission();
+        if(submission != null) {
+            try {
+                isReviewable = StringUtils.isNotEmpty(submission.getScheduleId()) || "2".equals(submission.getProtocolReviewTypeCode()) || "7".equalsIgnoreCase(submission.getProtocolReviewTypeCode()) && "112".equalsIgnoreCase(submission.getProtocolSubmissionType().getSubmissionTypeCode());
```
Comment (sskinner) [inline src/main/java/edu/arizona/kra/irb/onlinereview/CustomProtocolOnlineReviewServiceImpl.java]: Tons of magic strings, I'd suggest maybe spring props for this to keep it out of the code and easier to change; failing that, maybe made into constants for easy spotting at the top of the class; failing that, perhaps comments calling out what they are.

File: src/main/java/org/kuali/kra/irb/ProtocolForm.java [hunk 2]
```diff
@@ -106,17 +106,23 @@ protected String getDefaultDocumentTypeName() {
...
 
-        if (getProtocolDocument() != null && getProtocolDocument().getProtocol() != null) {
+        if (getProtocolDocument() != null && getProtocolDocument().getProtocol() != null && getProtocolDocument().getDocumentNumber() != null ) {
+            LOG.debug("getHeaderNavigationTabs(): getProtocolDocument().getDocumentNumber()= "+getProtocolDocument().getDocumentNumber());
+            LOG.debug("getHeaderNavigationTabs(): docHdrId= "+getProtocolDocument().getProtocol().getProtocolDocument().getDocumentNumber());
             String principalId = GlobalVariables.getUserSession().getPrincipalId();
             ProtocolSubmission submission = (ProtocolSubmission) getProtocolDocument().getProtocol().getProtocolSubmission();
             boolean isUserOnlineReviewer = onlineReviewService.isProtocolReviewer(principalId, false, submission);
-            boolean isUserIrbAdmin = getKraAuthorizationService().hasRole(GlobalVariables.getUserSession().getPrincipalId(), "KC-UNT", "IRB Administrator"); 
+            boolean isUserIrbAdmin = getKraAuthorizationService().hasRole(GlobalVariables.getUserSession().getPrincipalId(), "KC-UNT", "IRB Administrator");
```
Comment (sskinner) [inline src/main/java/org/kuali/kra/irb/ProtocolForm.java]: Are these magic strings likely to remain unchanged even during system upgrades?

File: src/main/java/org/kuali/kra/irb/actions/ActionHelper.java [hunk 3]
```diff
@@ -219,25 +214,10 @@ public void initializeProtocolActions() throws Exception {
...
-        protocolSuspendByDsmbBean = buildProtocolGenericActionBean(ProtocolActionType.SUSPENDED_BY_DSMB, 
-                Constants.PROTOCOL_SUSPEND_BY_DSMB_ACTION_PROPERTY_KEY);
-        protocolClosedBean = buildProtocolGenericActionBean(ProtocolActionType.CLOSED_ADMINISTRATIVELY_CLOSED,
-                Constants.PROTOCOL_CLOSE_ACTION_PROPERTY_KEY);
-        protocolPermitDataAnalysisBean = buildProtocolGenericActionBean(ProtocolActionType.DATA_ANALYSIS_ONLY, 
-                Constants.PROTOCOL_PERMIT_DATA_ANALYSIS_ACTION_PROPERTY_KEY);
-        protocolIrbAcknowledgementBean = buildProtocolGenericActionBean(ProtocolActionType.IRB_ACKNOWLEDGEMENT, 
-                Constants.PROTOCOL_IRB_ACKNOWLEDGEMENT_ACTION_PROPERTY_KEY);
+
+        //Removed initializing beans who have lazy instantiators...
```
Comment (sskinner) [inline src/main/java/org/kuali/kra/irb/actions/ActionHelper.java]: Nice!

File: src/main/java/edu/arizona/kra/irb/onlinereview/CustomProtocolOnlineReviewServiceImpl.java (hunk 1 above)
Comment (nataliac06) [reply src/main/java/edu/arizona/kra/irb/onlinereview/CustomProtocolOnlineReviewServiceImpl.java]: Yeah, just took the code from the parent service.. didn't bother switching those...

**Pull request merged**